hal/src/rtc/rtic/mod.rs

```diff
@@ -0,0 +1,335 @@
+//! | **Rollover period**                  | ~571 million years | ~17.8 million years |
+//! | **HPC interrupt period (SAMD11/21)** | 32 seconds         | 1 second            |
+//! | **HPC interrupt period (SAMx5x)**    | ~24 days           | ~18 hours           |
+//! | **Time resolution**                  | ~977 μs            | ~31 μs              |
+//!
+//! # Usage
+//!
+//! The monotonic should be created using the
+//! [macro](prelude::rtc_monotonic), which is included in the
+//! [prelude](crate::prelude). The first macro argument is the name of
+//! the global structure that will implement
+//! [`Monotonic`](rtic_time::Monotonic). The RTC clock rate must be
+//! known at compile time, and so the appropriate type from [`rtc_clock`] must
+//! be passed to the macro as the second argument.
+//!
+//! Sometime during initialization, the monotonic also must be started by
+//! calling the `start` method on the created monotonic. The
+//! [`Rtc`](crate::pac::Rtc) peripheral struct must be passed to `start` to
+//! ensure that the monotonic has complete control of the RTC.
+//!
+//! Note that the macro creates the RTC interrupt handler, and starting the
+//! monotonic enables RTC interrupts in the NVIC, so that this does not need to
+//! be done manually.
+//!
+//! # Example
+//!
+//! ```
+//! use atsamd_hal::prelude::*;
+//!
+//! // Create the monotonic struct named `Mono`
+//! rtc_monotonic!(Mono, rtc_clock::Clock32k);
+//!
+//! fn init() {
+//!     # // This is normally provided by the selected PAC
+//!     # let rtc = unsafe { core::mem::transmute(()) };
+//!     # let mut mclk = unsafe { core::mem::transmute(()) };
+//!     # let mut osc32kctrl = unsafe { core::mem::transmute(()) };
+//!     // Here the RTC clock source should be configured using the clocks API
+//!
+//!     // Start the monotonic
+//!     Mono::start(rtc);
+//! }
+//!
+//! async fn usage() {
+//!     loop {
+//!          // Use the monotonic
+//!          let timestamp = Mono::now();
+//!
+//!          Mono::delay_until(timestamp + 2u32.secs()).await;
+//!          Mono::delay(100u32.millis()).await;
+//!     }
+//! }
+//! ```
+//!
+//! # Other notes
+//!
+//! The number returned by
+//! [`Monotonic::now().ticks()`](rtic_monotonic::Monotonic::now) will always
+//! increase (barring monotonic rollover). However, due to the register
+//! [synchronization delay](https://onlinedocs.microchip.com/oxy/GUID-F5813793-E016-46F5-A9E2-718D8BCED496-en-US-14/GUID-0C52DB00-4BF6-4F41-85B5-B76529875364.html),
+//! the number returned may not always increment by one every time it changes.
+//! In fact, testing shows that it typically increments by four every time it
+//! changes. This is true regardless of the clock rate used, as the
+//! synchronization delay scales along with the clock period.
+
+/// Items for RTIC v1.
+///
+/// TODO: This probably needs to be modernized (e.g. it does not implement
+/// half-period counting) or deprecated/removed.
```

Comment:
I'm partial to adding a `#[deprecated]` attribute. Then we can yeet it at the next opportune occasion.

Reply:
If we choose to deprecate the item, its removal should be requested in #784.